src/main/java/com/epam/task/purchases/PurchaseArray.java

```diff
@@ -0,0 +1,63 @@
+package com.epam.task.purchases;
+
+import java.util.Arrays;
+import java.util.Comparator;
+
+public class PurchaseArray {
+    private final AbstractPurchase[] purchases;
+
+    public PurchaseArray(AbstractPurchase[] purchases) {
+        this.purchases = purchases;
+    }
+
+    public void sort(){
+        Arrays.sort(purchases);
+    }
+
+    public int search( Euro searchValue) {
+        Comparator<AbstractPurchase> comparator = AbstractPurchase::compareTo;
+
+        return Arrays.binarySearch(
+                purchases,
+                new DiscountIfMorePurchase(new Product(null, new Euro(searchValue)), 1, 0.10),
+                comparator);
+    }
```

Comment:
`AbstractPurchase` implements `Comparable`, it has internal comparator. 
```suggestion
    public int search( Euro searchValue) {
        return Arrays.binarySearch(
                purchases,
                new DiscountIfMorePurchase(new Product(null, new Euro(searchValue)), 1, 0.10));
    }
```